### PR TITLE
Injection of `test-hevm`

### PR DESCRIPTION
**Pull request by icetan**
To be able to build `solidityPackage`s with specific requirements on HEVM version to run tests.

### CODE REVIEW

File: nix/solidity-package.nix [hunk 1]
```diff
@@ -15,16 +15,20 @@
...
+    { doCheck ? true
+    , deps ? []
+    , solc ? pkgs.solc
+    , test-hevm ? pkgs.dapp2.test-hevm
+    , ...
+    } @ attrs:
       pkgs.stdenv.mkDerivation (rec {
-        buildInputs = [pkgs.dapp2.test-hevm solc];
+        inherit doCheck;
+        buildInputs = [test-hevm solc];
```
Comment (asymmetric) [inline nix/solidity-package.nix]: Can you add space after the brackets please?

File: nix/solidity-package.nix [hunk 2]
```diff
@@ -15,16 +15,20 @@
...
         passthru = {
           remappings = remappings deps;
           libPaths = libPaths deps;
         };
 
-        TEST = test;
```
Comment (asymmetric) [inline nix/solidity-package.nix]: Where was this being read?

File: nix/solidity-package.sh [hunk 3]
```diff
@@ -20,7 +20,10 @@ mkdir lib
 echo "$LIBSCRIPT" > setup.sh
 source setup.sh
 export DAPP_LIB=lib
-dapp2-test-hevm
+
+if [ "$doCheck" == 1 ]; then
```
Comment (asymmetric) [inline nix/solidity-package.sh]: Just to confirm: `true` is translated to 1? Do you know where this happens?

File: nix/solidity-package.nix (hunk 2 above)
Comment (icetan) [reply nix/solidity-package.nix]: Couldn't find it being read anywhere.

File: nix/solidity-package.nix (hunk 1 above)
Comment (icetan) [reply nix/solidity-package.nix]: Yes

File: nix/solidity-package.sh (hunk 3 above)
Comment (icetan) [reply nix/solidity-package.sh]: Yes, because `doCheck` gets set as an environment variable it's interpolated with `toString` which is `"1"` for `true`.

**Pull request merged**